rfcs/text/0013-multiple-files.md

```diff
@@ -0,0 +1,220 @@
+ECS has current support for multiple users in an event that was originally introduced in a previous RFC, this builds on the ideas introduced there, but for files.
+
+## Fields
+
+The new fields discussed in this RFC are the following:
+
+- `file.target.*`
+- `file.changes.*`
+
+Note that `file` would need to be made reusable with the following entries.
+
+```YAML
+# schemas/file.yml excerpt
+  reusable:
+    top_level: true
+    expected:
+      # Added for this RFC
+      - at: file
+        as: target
+      - at: file
+        as: changes
+```
+
+Currently `file` has 21 top-level fields and has a good number of fields that can be nested under it such as `code_signature`, `hash`, `pe`, and `x509`, so we may want to clarify exactly which fields we want to include (maybe just the top-level?).
+
+<!--
+Stage 3: Add or update all remaining field definitions. The list should now be exhaustive. The goal here is to validate the technical details of all remaining fields and to provide a basis for releasing these field definitions as beta in the schema. Use GitHub code blocks with yml syntax formatting.
+-->
+
+## Usage
+
+### File Modifications
+
+This can happen, for example if file permissions change, a file is renamed, or a file is written to (size/hash modification).
+
+Examples of data sources where this is applicable:
+
+- General FIM events
+- Syscall monitoring of `fchmod`-style calls
+- Windows Security log events including [#4670](https://www.ultimatewindowssecurity.com/securitylog/encyclopedia/event.aspx?eventID=4670)
+
+Here's an example where a file's permissions octet on Linux is changed from `0700` to `0755`:
+
+```JSON
+{
+  "file": {
+    "name": "test.txt",
+    "mode": "0700",
+    "changes": {
+      "mode": "0755"
+    }
+  }
+}
+```
+
+Here's an example of a file rename event;
+
+```JSON
+{
+  "file": {
+    "name": "abc.txt",
+    "changes": {
+      "name": "def.txt"
+    }
+  }
+}
+```
+
+Notice that `changes` only captures the fields that changed, not the fields that remain the same between modifications.
+
+Additionally, in some cases we're not always going to have access to the original field value (i.e. running `chmod 755 ...` isn't going to contain data about the original file permissions). Users should actively fill in all data that they can for the top-level fields, but, even if there is no corresponding top-level field, `changes` may contain the fields that are assumed to be modified.
+
+### File Copies
+
+This is a bit more complex than a modification. This can happen, for example if a file is duplicated via `cp` or some other utility, if a file is remotely transfered, or if a symlink or hardlink is created that references back to the original file. Unlike the modification events, I would imagine that copying would likely require most of the fields from the top-level of the file fields to be present in `target`, as file ownership, location, extensions, devices, timestamps, and names can all change during a copy event.
```

Comment:
For remotely transferred files, do you have any thoughts for the capturing of the remote host that'll be relevant to `file.target.*`? Or guidance on how to best capture remote copies using `file.target.*` plus other ECS fields?

Reply:
With regard to "capturing of the remote host" do you mean like hostnames/ips and such? If think that leveraging the `source`/`destination` or `client`/`server` fields (depending on protocol/modeling) would probably make sense for something like that. Do you want me to add that to the doc?

Reply:
Yeah, please do.